base/arraymath.jl

```diff
@@ -89,7 +89,7 @@ function _elementwise{T}(op, ::Type{T}, A::AbstractArray, B::AbstractArray)
     return F
 end
 
-for f in (:div, :mod, :rem, :&, :|, :xor, :/, :\, :*, :+, :-)
+for f in (:mod, :rem, :&, :|, :xor, :/, :\, :*, :+, :-)
```

Comment:
(Likewise here, this change moves `div`'s promotion behavior in the direction of #19669.)